packages/utils/src/ast.test.ts

```diff
@@ -0,0 +1,247 @@
+import { getAST, getCode, postProcessAST } from './ast';
+
+describe('getAST', () => {
+  it('returns an AST for the provided code', () => {
+    const code = `const foo = 'bar'; console.log(foo);`;
+
+    // eslint-disable-next-line jest/no-restricted-matchers
+    expect(getAST(code)).toMatchSnapshot();
+  });
+
+  it(`doesn't attach comments to the AST if configured`, () => {
+    const code = `
+      // This is a comment.
+      const foo = 'bar';
+    `;
+
+    const ast = getAST(code, false);
+    expect(getCode(ast)).not.toContain('// This is a comment.');
+  });
+
+  it('forwards parser errors', () => {
+    // Invalid code
+    const code = `const foo bar;`;
+
+    expect(() => getAST(code)).toThrow(
+      `'Const declarations' require an initialization value.`,
+    );
+  });
+});
+
+describe('postProcessAST', () => {
+  it('wraps eval', () => {
+    const code = `
+      eval(bar);
+      foo.eval(bar, baz);
+      foo.bar.eval(baz, qux);
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(`
+      "(1, eval)(bar);
+      (1, foo.eval)(bar, baz);
+      (1, foo.bar.eval)(baz, qux);"
+    `);
+  });
+
+  it('removes the Buffer argument', () => {
+    const code = `
+      (function (Buffer, foo) {
+        const bar = 'baz';
+      });
+
+      function foo(Buffer) {
+        const bar = 'baz';
+      }
+    `;
+
+    const ast = getAST(code);
+    const processedCode = postProcessAST(ast);
+
+    expect(getCode(processedCode)).toMatchInlineSnapshot(`
+      "(function (foo) {
+        const bar = \\"baz\\";
+      });
+
+      function foo(Buffer) {
+        const bar = \\"baz\\";
+      }"
+    `);
+  });
+
+  it('inserts the regenerator runtime global when used', () => {
```

Comment:
I wonder if there is a better way to do this still 🤔  Wouldn't it change behaviour to define it for each scope?